tests/main/snap-remove-terminate/task.yaml

```diff
@@ -0,0 +1,35 @@
+summary: Ensure that the remove --terminate flag kills running snap apps.
+
+details: |
+    This test spawns a snap app background process and checks that it
+    gets terminated when removing with the --terminate flag.
+
+systems:
+    # Ubuntu 14.04's special version of systemd doesn't have StartTransientUnit API.
+    - -ubuntu-14.04-*
+
+prepare: |
+    snap install test-snapd-sh
+
+    # TODO: Discard api-client when --terminate flag is added to remove command
+    "$TESTSTOOLS"/snaps-state install-local api-client
+    snap connect api-client:snapd-control
+
+restore: |
+    snap remove --purge api-client
+
+execute: |
+    echo "Start a long running process"
+    test-snapd-sh.sh -c "sleep 100000" &
```

Comment:
We have some tests that use `systemd-run` instead of hand-holding the process like that. There were countless issues with odd race conditions and stuck file descriptors with code running like that, using shell job control primitives.